katana/framework.py

```diff
@@ -15,74 +41,73 @@ class Katana():
-            zone_letter:    UTM zone letter (probably N)
-            zone_number:    UTM zone letter
-            buff:           WindNinja domain buffer desired (m)
-            start_date:     datetime object for start date
-            end_date:       datetime object for end date
-            directory:      directory containing HRRR grib2
-                            files (directory/hrrr.<date>/hrrr*.grib2)
-            out_dir:        output directory where (out_dir/data<date>)
-                            will be written
-            wn_cfg:         file path where WindNinja config file
-                            will be stored
-            nthreads:       number of threads used to run WindNinja
-            nthreads_w:     number of threads used for wgrib2 commands
-            dxy:            grid resolution for running WindNinja
-            loglevel:       level for logging info
-            logfile:        file where log will be stored
-            make_new_gribs: option to use existing gribs if this
-                            step has been completed
+    def __init__(self, config):
+        """[summary]
+
+        Arguments:
+            config {string} -- path to the config file or an
+                                inicheck UserConfig object
         """
 
-        self.start_timing = datetime.now()
-        ################################################
-        # Start parsing the arguments
-        ################################################
-        self.fp_dem = fp_dem
-        self.zone_letter = zone_letter
-        self.zone_number = zone_number
-        self.buff = buff
+        if isinstance(config, str):
+            if not os.path.isfile(config):
+                raise Exception('Configuration file does not exist --> {}'
+                                .format(config))
 
-        # find start and end dates
-        self.start_date = start_date
-        self.end_date = end_date
+            try:
+                mcfg = MasterConfig(modules=['katana'])
 
-        self.fmt_date = '%Y%m%d'
-        self.directory = directory
-        self.out_dir = out_dir
+                # Read in the original users config
+                self.ucfg = get_user_config(config, mcfg=mcfg)
+                self.config_file = config
+
+            except UnicodeDecodeError as e:
+                print(e)
+                raise Exception(('The configuration file is not encoded in '
+                                 'UTF-8, please change and retry'))
 
-        self.nthreads_w = nthreads_w
+        elif isinstance(config, UserConfig):
+            self.ucfg = config
+            self.config_file = ''
+
+        else:
+            raise Exception(
+                'Config passed to Katana is neither file name nor \
+                    UserConfig instance')
 
-        self.make_new_gribs = make_new_gribs
+        warnings, errors = check_config(self.ucfg)
+        print_config_report(warnings, errors)
```

Comment:
Consider exit if errors != []?

Reply:
@micahjohnson150 We should think about adding this to `inicheck`, maybe a `print_config_report_error` or a different function that makes it a command in `inicheck`. Then it's not up to the user to implement this and it can all be handled in `inicheck`

Reply:
or raise an exception

Reply:
The reason I don't is because the definition of errors is left to the developer since they choose between warnings and errors by choosing types and since they can make types they also can specify what constitutes an error. Which seems like it would always be something you would want to exit for but I can't be sure.

I personally would like to leave it to the end-use/developer to decide whether to error out or not.